modules/custom_operations/user_ie_extensions/fft.cpp

```diff
@@ -0,0 +1,366 @@
+        cvInitMatHeader(srcBot, botH, width, CV_32FC2, data + topH * width * 2, step);
+        cvInitMatHeader(dstTop, topH, width, CV_32FC2, data + botH * width * 2, step);
+        cvInitMatHeader(dstBot, botH, width, CV_32FC2, data, step);
+
+        CvMat* tmp = cvCloneMat(srcTop);
+        cvCopy(srcBot, dstBot, 0);
+        cvCopy(tmp, dstTop, 0);
+
+        cvReleaseMat(&tmp);
+        delete srcTop;
+        delete srcBot;
+        delete dstTop;
+        delete dstBot;
+
+        // Swap columns.
+        CvMat* srcL = new CvMat();
+        CvMat* srcR = new CvMat();
+        CvMat* dstL = new CvMat();
+        CvMat* dstR = new CvMat();
+        int leftW = inverse ? w2 : (w2 + width % 2);
+        int rightW = width - leftW;
+
+        cvInitMatHeader(srcL, height, leftW, CV_32FC2, data, step);
+        cvInitMatHeader(srcR, height, rightW, CV_32FC2, data + leftW * 2, step);
+        cvInitMatHeader(dstL, height, leftW, CV_32FC2, data + rightW * 2, step);
+        cvInitMatHeader(dstR, height, rightW, CV_32FC2, data, step);
+
+        tmp = cvCloneMat(srcL);
+        cvCopy(srcR, dstR, 0);
+        cvCopy(tmp, dstL, 0);
+
+        cvReleaseMat(&tmp);
+        delete srcL;
+        delete srcR;
+        delete dstL;
+        delete dstR;
+
+        return;
+    }
+
+    CvMat* tl = new CvMat();
+    CvMat* tr = new CvMat();
+    CvMat* bl = new CvMat();
+    CvMat* br = new CvMat();
+
+    cvInitMatHeader(tl, h2, w2, CV_32FC2, data, step);
+    cvInitMatHeader(tr, h2, w2, CV_32FC2, data + width, step);
+    cvInitMatHeader(bl, h2, w2, CV_32FC2, data + height * width, step);
+    cvInitMatHeader(br, h2, w2, CV_32FC2, data + height * width + width, step);
+
+    CvArr* mask = 0;
+    CvMat* tmp = cvCloneMat(tl);
+    cvCopy(br, tl, mask);
+    cvCopy(tmp, br, mask);
+
+    cvCopy(tr, tmp, mask);
+    cvCopy(bl, tr, mask);
+    cvCopy(tmp, bl, mask);
+
+    cvReleaseMat(&tmp);
+
+    delete tl;
+    delete tr;
+    delete bl;
+    delete br;
+}
+
+//! [op:ctor]
+FFT::FFT(const ov::OutputVector& args, bool inverse, bool centered) : Op(args) {
+    loadOpenCV();
+    constructor_validate_and_infer_types();
+    this->inverse = inverse;
+    this->centered = centered;
+}
+//! [op:ctor]
+
+//! [op:validate]
+void FFT::validate_and_infer_types() {
+    auto outShape = get_input_partial_shape(0);
+    set_output_type(0, get_input_element_type(0), outShape);
+}
+//! [op:validate]
+
+//! [op:copy]
+std::shared_ptr<ov::Node> FFT::clone_with_new_inputs(const ov::OutputVector& new_args) const {
+    OPENVINO_ASSERT(new_args.size() == 2, "Incorrect number of new arguments");
+    return std::make_shared<FFT>(new_args, inverse, centered);
+}
+//! [op:copy]
+
+//! [op:visit_attributes]
+bool FFT::visit_attributes(ov::AttributeVisitor& visitor) {
+    int inverse_i = static_cast<int>(inverse);
+    int centered_i = static_cast<int>(centered);
+    visitor.on_attribute("inverse", inverse_i);
```

Comment:
Why do you need to have static_cast to int?
Visit attributes supports boolean variables from the box.

Reply:
There were issues with boolean attributes (always false) when switched to 2022.1

Reply:
Is it issue inside the 2022.1 release? I don't think that we need to support old non-LTS release... Master branch should contain the code without WAs which are needed to support old releases